Drivers/Src/MotorController.c

```diff
@@ -1,124 +1,217 @@
+static OS_SEM	MotorController_ReceiveSem4;
+static float CurrentVelocity = 0;
+
+static tritium_error_code_t Motor_FaultBitmap = T_NONE;
+
+static bool is_initialized = false;
+
+/**
+ * @brief   Assert Error if Tritium sends error. When Fault Bitmap is set,
+ *          and semaphore is posted, Fault state will run.
+ * @param   motor_err Bitmap which has motor error codes
+ */
+static void assertTritiumError(tritium_error_code_t motor_err){
+    OS_ERR err;
+    if(motor_err != T_NONE){
+        FaultBitmap.Fault_TRITIUM = 1;
+        OSSemPost(&FaultState_Sem4, OS_OPT_POST_1, &err);
+        assertOSError(0, err);
+    }
+}
+
+/**
+ * @brief   Releases hold of the mailbox semaphore.
+ * @note	Do not call directly.
+ */
+static void MotorController_Release(void) {
+	OS_ERR err;
+
+	OSSemPost(&MotorController_MailSem4,
+			  OS_OPT_POST_1,
+			  &err);
+	assertOSError(0, err);
+}
+
+/**
+ * @brief	Increments the receive semaphore.
+ * @note	Do not call directly.
+ */
+static void MotorController_CountIncoming(void) {
+	OS_ERR err;
+
+	OSSemPost(&MotorController_ReceiveSem4,
+			  OS_OPT_POST_1,
+			  &err);
+	assertOSError(0, err);
+}
 
 /**
  * @brief   Initializes the motor controller
- * @param   None
+ * @param   busCurrentFractionalSetPoint fraction of the bus current to allow the motor to draw
  * @return  None
  */ 
-void MotorController_Init(){
-    BSP_CAN_Init(CAN_3);
+void MotorController_Init(float busCurrentFractionalSetPoint){
+    if (is_initialized) return;
+    is_initialized = true; // Ensure that we only execute the function once
+    CPU_TS ts;
+	OS_ERR err;
+    OSSemCreate(&MotorController_MailSem4,
+                "Motor Controller Mailbox Semaphore",
+                3,	// Number of mailboxes
+                &err);
+	assertOSError(0, err);
+
+    OSSemCreate(&MotorController_ReceiveSem4,
+                "Motor Controller Receive Semaphore",
+                0,	// Number of mailboxes
+                &err);
+	assertOSError(0, err);
+
+    BSP_CAN_Init(CAN_3, MotorController_CountIncoming, MotorController_Release);
+
+    uint8_t data[8] = {0};
+    memcpy(
+        data,
+        &busCurrentFractionalSetPoint,
+        sizeof(busCurrentFractionalSetPoint)
+    );
+    OSSemPend(&MotorController_MailSem4,
+            0,
+            OS_OPT_PEND_BLOCKING,
+            &ts,
+            &err);
+	assertOSError(0, err);
+    ErrorStatus initCommand = BSP_CAN_Write(CAN_3, MOTOR_POWER, data, MAX_CAN_LEN);
+    if (initCommand == ERROR) {
+		MotorController_Release();
+        Motor_FaultBitmap = T_INIT_FAIL;
+        assertTritiumError(Motor_FaultBitmap);
+	}
 }
 
 /**
- * @brief   Sends MOTOR DRIVE command on CAN2
+ * @brief   Sends MOTOR DRIVE command on CAN3
  * @param   newVelocity desired motor velocity setpoint in m/s
- * @param   motorCurrent desired motor current setpoint as a percentage of max current setting
+ * @param   motorCurrent desired motor current setpoint as a percentage of max current setting (0.0-1.0)
```

Comment:
Do you want any parameter checking in this function? What happens if you get passed an invalid parameter (and is this possible)?